docs/runbooks/KubeVirtVMGuestMemoryHigh.md

```diff
@@ -0,0 +1,225 @@
+its available memory for more than 5 minutes, as seen from the guest OS
+perspective.
+
+## Impact
+
+High memory usage in a VM can lead to:
+- **Performance degradation** - Applications may become slow due to memory pressure
+- **Memory swapping** - Guest OS starts using swap space, severely impacting performance
+- **Application crashes** - Out-of-memory (OOM) conditions may kill processes
+- **System instability** - Critical system processes may fail due to memory exhaustion
+- **Cascading failures** - Memory pressure can affect the entire application stack
+
+## Diagnosis
+
+### 1. Identify the Affected VM
+
+The alert provides the following information: VM Name, Namespace, Node and
+Memory Usage.
+
+### 2. Check VM Memory Allocation
+
+```bash
+# Get VM configuration
+kubectl get vm <vm-name> -n <namespace> -o yaml
+
+# Check current VMI status
+kubectl get vmi <vm-name> -n <namespace> -o yaml
+```
+
+Look for:
+- `spec.domain.memory.guest` - Guest memory allocation
+- `spec.domain.resources.requests.memory` - Memory resource requests
+- `spec.domain.resources.limits.memory` - Memory resource limits
+
+### 3. Monitor Current Memory Usage
+
+```bash
+# Check VM resource consumption
+virtctl top vmi <vm-name> -n <namespace>
+
+# Get detailed memory metrics
+kubectl exec -n <namespace> \
+  $(kubectl get pods -n <namespace> -l kubevirt.io/vmi=<vm-name> \
+    -o jsonpath='{.items[0].metadata.name}') \
+  -c compute -- cat /proc/meminfo
+```
+
+### 4. Check Memory Usage Inside the VM
+
+Connect to the VM to identify memory consumption:
+
+```bash
+# Console access
+virtctl console <vm-name> -n <namespace>
+
+# Or SSH if available
+virtctl ssh <vm-name> -n <namespace>
+```
+
+Inside the VM, run:
+```bash
+# Overall memory usage
+free -h
+
+# Top processes by memory usage
+top -o %MEM
+
+# Detailed memory information
+cat /proc/meminfo
+
+# Check for memory-hungry processes
+ps aux --sort=-%mem | head -20
+
+# Check for memory leaks
+pmap -x $(pgrep -f <process-name>)
+```
+
+### 5. Check for Memory Pressure Indicators
+
+```bash
+# Check for OOM events in VM
+dmesg | grep -i "killed process"
+journalctl | grep -i "out of memory"
+
+# Check swap usage
+swapon --show
+free -h | grep Swap
+
+# Check memory pressure stall information (if available)
+cat /proc/pressure/memory
+```
+
+### 6. Check Node-Level Metrics
+
+```bash
+# Check node memory usage
+kubectl top node <node>
+
+# Check other VMs on the same node
+kubectl get vmi --all-namespaces --field-selector spec.nodeName=<node>
+
+# Check node conditions and pressure
+kubectl describe node <node>
+```
+
+### 7. Historical Analysis
+
+Use Prometheus/Grafana to analyze:
+
+```promql
+# VM memory usage over time
+kubevirt_vmi_memory_used_bytes{name="<vm-name>", namespace="<namespace>"}
+
+# VM memory usage percentage
+(kubevirt_vmi_memory_used_bytes{name="<vm-name>", namespace="<namespace>"} / \
+ kubevirt_vmi_memory_available_bytes{name="<vm-name>", namespace="<namespace>"}) * 100
```

Comment:
this metric mentioned that "This value may not be accurate if a balloon driver is in use or if the guest OS does not initialize all assigned pages" might want to point it also in this runbook